views/js/component/activityMonitoring/activityGraph.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * @author Aleh Hutnikau <hutnikau@1pt.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'ui/component',
+    'util/url',
+    'c3',
+    'd3'
+], function ($, _, __, component, url, c3) {
+    'use strict';
+
+    /**
+     * Some default config
+     * @type {Object}
+     * @private
+     */
+    var _defaults = {
+        autoRefresh : 0,
+        autoRefreshBar : false,
+        transition: {
+            duration: 3000
+        },
+        graphConfig : {
+            padding: {
+                bottom: 0,
+                left: 35,
+            },
+            data: {
+                x: 'time',
+                xFormat: '%Y-%m-%d %H:%M:%S',
+                mimeType: 'json',
+                type: 'bar'
+            },
+            axis: {
+                x: {
+                    type: 'timeseries',
+                    tick: {
+                        format: '%H:%M',
+                    },
+                    label: {
+                        position: 'bottom center'
+                    }
+                },
+                y: {
+                    label: {
+                        position: 'top'
+                    }
+                }
+            },
+        }
+    };
+
+    /**
+     * @param {jQuery} $refreshProgress
+     * @param duration
+     */
+    function runRefreshProgressBar($refreshProgress, duration) {
+        $refreshProgress.stop();
+        $refreshProgress.css({width:'0%'});
+        $refreshProgress.animate({
```

Comment:
would be better to try using a dynamically set css transition instead. this will greatly improve performance as the animation will be delegated to the GPU with css transition